src/test/java/org/folio/inventory/validation/experimental/TargetItemStatusValidatorTest.java

```diff
@@ -0,0 +1,93 @@
+package org.folio.inventory.validation.experimental;
+
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import lombok.SneakyThrows;
+import org.folio.inventory.domain.items.Item;
+import org.folio.inventory.domain.items.ItemStatusName;
+
+import org.folio.inventory.domain.items.Status;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.folio.inventory.domain.items.ItemStatusName.IN_PROCESS;
+import static org.folio.inventory.domain.items.ItemStatusName.forName;
+
+@RunWith(JUnitParamsRunner.class)
+public class TargetItemStatusValidatorTest {
+  private static TargetItemStatusValidator validator;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    validator = new TargetItemStatusValidator();
+  }
+
+  @Parameters(
+    {
+      "In process",
+      "In process (non-requestable)"
+    }
+  )
+  @Test
+  public void targetStatusAllowMarkAs(String statusName) {
+    final var targetItemStatusName = forName(statusName);
+    final var statusValidator = validator.getValidator(targetItemStatusName);
+    assertThat(statusValidator).isNotNull();
+    final var allowedStatuses = statusValidator.getAllStatusesAllowedToMark();
+    System.out.println("Allowed statuses for:"+statusValidator.getItemStatusName());
+    allowedStatuses.stream().forEach(x -> {
+      System.out.println("\t"+x);
+      Item item = new Item(null, null, new Status(x), null, null, null);
+      assertThat(statusValidator.isItemAllowedToMark(item)).isTrue();
+    });
+
+    Set<ItemStatusName> disallowedStatuses = new HashSet<>();
+    Collections.addAll(disallowedStatuses,ItemStatusName.values());
+    disallowedStatuses.removeAll(allowedStatuses);
+    System.out.println("Disallowed statuses for:"+statusValidator.getItemStatusName());
+    disallowedStatuses.stream().forEach(x -> {
+      System.out.println("\t"+x);
+      Item item = new Item(null, null, new Status(x), null, null, null);
+      assertThat(statusValidator.isItemAllowedToMark(item)).isFalse();
+    });
+  }
+
+  @SneakyThrows
+  @Parameters({
+    "Available",
+    "In transit",
+    "Awaiting pickup",
+    "Missing",
+    "Withdrawn",
+    "Lost and paid",
+    "Paged",
+    "Awaiting delivery",
+    "Order closed"
+  })
+  @Test
+  public void itemCanBeMarkedAsInProcessWhenInAcceptableSourceStatus(String sourceStatus) {
```

Comment:
I think there needs to be a corresponding negative test for the source statuses which aren't allowed to be marked as in process.